src/main/kotlin/net/atos/zac/app/informatieobjecten/EnkelvoudigInformatieObjectRestService.kt

```diff
@@ -629,16 +629,15 @@ class EnkelvoudigInformatieObjectRestService @Inject constructor(
         }
 
     private fun listEnkelvoudigInformatieobjectenVoorZaak(zaak: Zaak): MutableList<RESTEnkelvoudigInformatieobject> =
-        zaak
-            .let(zrcClientService::listZaakinformatieobjecten)
-            .let(informatieobjectConverter::convertToREST)
+        zaak.let(zrcClientService::listZaakinformatieobjecten)
+            .map(informatieobjectConverter::convertToREST)
+            .toMutableList()
 
     private fun listGekoppeldeZaakEnkelvoudigInformatieobjectenVoorZaak(
         zaakURI: URI,
         relatieType: RelatieType
     ): List<RESTGekoppeldeZaakEnkelvoudigInformatieObject> =
-        zaakURI
-            .let(zrcClientService::readZaak)
+        zaakURI.let(zrcClientService::readZaak)
             .let { zaak ->
```

Comment:
Map should be split out of the `let` statement:
```
    .let(zrcClientService::listZaakinformatieobjecten)
    .map { informatieobjectConverter.convertToREST(it, relatieType, zaak) }
    .toList()
```

Reply:
Hmm, I don't think so because the `zaak` is needed in that `map`. This code you suggest does not compile because of that.